### PR TITLE
Editable bread crumb bar for RTDB/Firestore

### PR DESCRIPTION
**Pull request by tstirrat**
Firestore:

Screen Shot 2020-03-20 at 12 21 14 PM

Edit:
Screen Shot 2020-03-20 at 12 21 19 PM

RTDB, includes children as actions on right:
Screen Shot 2020-03-20 at 12 27 30 PM
Screen Shot 2020-03-20 at 12 27 35 PM

### CODE REVIEW

**Comment by tstirrat** [comment]
Added inputPrefix:

Screen Shot 2020-03-20 at 1 28 59 PM

File: src/components/Firestore/index.tsx [hunk 1]
```diff
@@ -68,6 +67,11 @@ export const Firestore: React.FC<Props> = ({ config, projectId }) => {
     api.nukeDocuments();
   }
 
+  function handleNavigate(path: string) {
+    // TODO: needs a better impl
+    window.location.href = `/firestore/${path}`;
```
Comment (yuchenshi) [inline src/components/Firestore/index.tsx]: Please render a `<Redirect>` or call `history.pushState` (with `useHistory()` I believe).

File: src/components/Database/Database.tsx [hunk 2]
```diff
@@ -53,16 +60,21 @@ export const Database: React.FC<Props> = ({ config, namespace }) => {
...
-          />
-        )}
-      </CardActionBar>
       {ref ? (
         <>
+          <InteractiveBreadCrumbBar
+            base={`/database/${namespace}/data`}
+            path={new URL(ref.toString()).pathname}
+            inputPrefix={getPrefix(ref)}
+            onNavigate={console.log.bind(null, 'onNavigate')}
```
Comment (yuchenshi) [inline src/components/Database/Database.tsx]: Can we have a TODO comment here please?

File: src/components/Firestore/index.tsx (hunk 1 above)
Comment (tstirrat) [reply src/components/Firestore/index.tsx]: Ah! wasn't sure the best way, thanks.

File: src/components/Database/Database.tsx (hunk 2 above)
Comment (tstirrat) [reply src/components/Database/Database.tsx]: Updated with a real implementation

File: src/components/Firestore/index.tsx (hunk 1 above)
Comment (tstirrat) [reply src/components/Firestore/index.tsx]: Done.

**Pull request merged**